core/services/hash_utils.py

```
@@ -0,0 +1,78 @@
+    
+    Args:
+        mobile: Mobile number with country code
+        hmac_secret: Secret key for HMAC
+        hash_length: Output hash length (default 8)
+        server_timestamp: Override timestamp (for testing)
+    
+    Returns:
+        (hash: str, generated_at: datetime)
+    """
+    if server_timestamp is None:
+        server_timestamp = datetime.utcnow()
```

Ruff (0.14.10)

`datetime.datetime.utcnow()` used

(DTZ003)